core/src/main/java/com/microsoft/applicationinsights/TelemetryClient.java

```diff
@@ -95,10 +95,10 @@ public TelemetryClient() {
      * application session.
      */
     public TelemetryContext getContext() {
-        if (context == null || (context.getInstrumentationKey() != null &&  !context.getInstrumentationKey().equals(configuration.getInstrumentationKey()))) {
+        if (context == null || (configuration.getInstrumentationKey() != null &&  !configuration.getInstrumentationKey().equals(context.getInstrumentationKey()))) {
```

Comment:
this is the one liner fix.. just switch context with configuration.

Reply:
TelemetryClient.context needs to be updated after the connection string is lazily set.  Global telemetry client gets updated but not for new TelemetryClient()